docs/new-architecture-library-intro.md

```diff
@@ -71,15 +100,35 @@ export default (codegenNativeComponent<NativeProps>(
 ): HostComponent<NativeProps>);
 ```
 
-### Supported Flow Types
+</TabItem>
+<TabItem value="typescript">
+
+```ts
+import type { ViewProps } from 'ViewPropTypes';
+import type { HostComponent } from 'react-native';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
+
+export interface NativeProps extends ViewProps {
+  // add other props here
+}
+
+export default codegenNativeComponent<NativeProps>(
+  '<FABRIC COMPONENT>'
+) as HostComponent<NativeProps>;
+```
+
+</TabItem>
+</Tabs>
+
+### Supported Types
 
-When using Flow, you will be using [type annotations](https://flow.org/en/docs/types/) to define your spec. Keeping in mind that the goal of defining a JavaScript spec is to ensure the generated native interface code is type safe, the set of supported Flow types will be those that can be mapped one-to-one to a corresponding type on the native platform.
+When using Flow or TypeScript, you will be using [type annotations](https://flow.org/en/docs/types/) to define your spec. Keeping in mind that the goal of defining a JavaScript spec is to ensure the generated native interface code is type safe, the set of supported types will be those that can be mapped one-to-one to a corresponding type on the native platform.
 
 <!-- alex ignore savage -->
 
-In general, this means you can use primitive types (strings, numbers, booleans), as well as function types, object types, and array types. Union types, on the other hand, are not supported. All types must be read-only in Flow: either `+` or `$ReadOnly<>` or `{||}` objects.
+In general, this means you can use primitive types (strings, numbers, booleans), as well as function types, object types, and array types. Union types, on the other hand, are not supported. All types must be read-only. For Flow: either `+` or `$ReadOnly<>` or `{||}` objects. For TypeScript: `readonly` for properties, `Readonly<>` for objects, and `ReadonlyArray<>` for arrays.
 
-> See Appendix [I. Flow Type to Native Type Mapping](#i-flow-type-to-native-type-mapping).
+> See Appendix [I. Spec Type to Native Type Mapping](#i-spec-type-to-native-type-mapping).
```

Comment:
@cortinico Noticed this doesn't exist. Should we put a `TODO:` or something here?

Reply:
Sorry I missed this ping.

> Should we put a TODO: or something here?

I would not put a TODO in the doc. Just mention that the TypeScript section will be added afterwards. I also believe the table is sort of trivial so it's something we can probably do also on our end (@cipolleschi how do you feel about that) if you don't have much capacity. That would make all the TS doc ready end to end 👍

Reply:
@cortinico this should be good to go now